src/pages/UserPages/UserWallet/Wallet.jsx

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from 'react'
+import { Form, FormField, FormUpload, SubmitButton } from '../../../components/Form'
+import BaseModal from '../../../components/Modal/BaseModal'
+import { thousands_separators } from '../../../utils/helperFunctions'
+import * as Yup from 'yup'
+import { useDispatch, useSelector } from 'react-redux'
+import { shallowEqual } from 'react-redux'
+import { disableButton } from '../../../store/reducers/buttonState'
+import { sendTransaction, updateWallet } from '../../../store/reducers/user'
+
+
+
+
+
+const Wallet = ({ wallet, uid, walletTransactions }) => {
+
+
+    const[showTopUp, setShowTopUp] = useState(false)
+    const[receiptNeeded, setReceiptNeeded] = useState(true)
+
+    const{ buttonDisable } = useSelector(state => ({
+        buttonDisable : state.buttonState.buttonDisable
+    }), shallowEqual)
+
+    const dispatch = useDispatch()
+
+    const creditFunc = () => {
+        setShowTopUp(true)
+        setReceiptNeeded(true)
+    }
+
+    const debitFunc = () => {
+        setShowTopUp(true)
+        setReceiptNeeded(false)
+    }
+
+
+    const validationSchema = Yup.object().shape({
+        amount: receiptNeeded ? Yup.number().min(1000).max(1000000).required().label("Credit Amount") : Yup.number().min(1000).max(wallet.available).required().label("Debit Amount"),
+        receipt: receiptNeeded && Yup.string().required().label("Receipt/Teller Image")
+    })
+
+
+    const submitForm = ({ amount, receipt }) => {
+
+        const transactionData = {
+            txType : receiptNeeded ? "credit" : "debit",
+            uid,
+            amount,
+            receipt : receiptNeeded ? receipt : "",
+            walletTransactions
+        }
+
+        const newWallet = {
+            pendingCredit : receiptNeeded ? wallet.pendingCredit + amount : wallet.pendingCredit,
+            pendingDebit : receiptNeeded ? wallet.pendingDebit : wallet.pendingDebit + amount,
+            available : receiptNeeded ? wallet.available : wallet.available - amount
+        }
+
+        dispatch(sendTransaction(transactionData))
+        dispatch(disableButton())
```

Comment:
This should be the first thing to be done in this function, the button needs t o be disabled before any computation is actually done.

Reply:
This function isn't working as expected. I do not know why, I tried login sample data to see but it still didn't work at a point it did work but the data was only created in the `transactions` collection but the `user` document wasn't updated.

Reply:
yes...there's gonna be a delay bcos of the sleeping cloud function.

Since user cannot directly update stuff in the private subcollection, we are using cloud functions to do so... so the cloud function triggers when the user creates a transaction. Since we aren't hitting the cloud function so frequently, sometimes it goes to sleep to save computing power. So It'll take a while to wake up. but as long as you saw the transaction go in... it's working.


I will reverse the buttonDisable function.